v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandler.java

```diff
@@ -0,0 +1,466 @@
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link Float} object containing the value represented in cassandra type.
+     */
+    public static Float handleCassandraFloatType(String colName, JSONObject valuesJson) {
+        return valuesJson.getBigDecimal(colName).floatValue();
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link Double} object containing the value represented in cassandra type.
+     */
+    public static Double handleCassandraDoubleType(String colName, JSONObject valuesJson) {
+        return valuesJson.getBigDecimal(colName).doubleValue();
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link ByteBuffer} object containing the value represented in cassandra type.
+     */
+    public static ByteBuffer handleCassandraBlobType(String colName, JSONObject valuesJson) {
+        Object colValue = valuesJson.opt(colName);
+        if (colValue == null) {
+            return null;
+        }
+        return parseBlobType(colName, colValue);
+
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param colValue - contains all the key value for current incoming stream.
+     *
+     * @return a {@link ByteBuffer} object containing the value represented in cassandra type.
+     */
+    private static ByteBuffer parseBlobType(String colName, Object colValue) {
+        byte[] byteArray;
+
+        if (colValue instanceof byte[]) {
+            byteArray = (byte[]) colValue;
+        } else if (colValue instanceof String) {
+            byteArray = java.util.Base64.getDecoder().decode((String) colValue);
+        } else {
+            throw new IllegalArgumentException("Unsupported type for column " + colName);
+        }
+
+        return ByteBuffer.wrap(byteArray);
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link Date} object containing the value represented in cassandra type.
+     */
+    public static Date handleCassandraDateType(String colName, JSONObject valuesJson) {
+        return handleCassandraGenericDateType(colName, valuesJson, "yyyy-MM-dd");
```

Comment:
I might be missing the complete context here.

In case this is going to be passed to Cassadra 4.0 Driver later on,
Would it make sense to [Local Date](https://docs.datastax.com/en/drivers/java/4.1/com/datastax/oss/driver/api/core/data/SettableById.html#setLocalDate-com.datastax.oss.driver.api.core.CqlIdentifier-java.time.LocalDate-)

For Some of the temporal types, we might need to be careful of the differences in how time zones are handled in Spanner and Cassandra.

For example, Cassandra Date is free of time zone.

Reply:
Yes @VardhanThigle we are using the LocalDate here

Reply:
Thanks. Looks like this got addressed as a part of [pull/2054/commits/5f9affa3566411e196cafe420fb55fc918d20a0a](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/2054/commits/5f9affa3566411e196cafe420fb55fc918d20a0a). Resolving.